src/builtins.rs

```diff
@@ -56,7 +61,44 @@ pub fn export(args: Vec<&str>) -> Result<String, Box<dyn Error>> {
         return Err("export: too many arguments".into());
     } else {
         let (key, value) = args.first().unwrap().split_once("=").unwrap();
-        unsafe { std::env::set_var(key, value) };
+        std::env::set_var(key, value);
         Ok("".to_string())
     }
 }
+
+pub fn alias(args: Vec<&str>) -> Result<String, Box<dyn Error>> {
+    if args.len() != 1 {
+        return Err("alias: not enough arguments or too many arguments".into());
+    }
+
+    let alias_input = args[0];
+    let parts: Vec<&str> = alias_input.splitn(2, '=').collect();
+
+    if parts.len() != 2 {
+        return Err("alias: invalid format, expected name=value".into());
+    }
+
+    let alias_name = parts[0].to_string();
+    let alias_value = parts[1].to_string();
+
+    unsafe {
+        if ALIASES.is_none() {
+            ALIASES = Some(HashMap::new());
+        }
+        ALIASES.as_mut().unwrap().insert(alias_name, alias_value);
+    }
+    Ok("".to_string())
+}
+
+pub fn expand_aliases(input: &str) -> String {
```

Comment:
This only detects aliases if they are the first word in a statement. Just realized that this isn't what we want I don't think because we also want aliases in sequences (and maybe pipelines?) to work as well I think:

```bash
alias_1 arg1; alias_2;
```

So this method should probably work after the tokenization step in `main.rs`.